src/gobprepare/prepare_client.py

```diff
@@ -254,6 +261,72 @@ def action_check_row_counts(self, action: RowCountConfig) -> bool:
                 )
         return tables_counts_ok
 
+    def action_check_source_sync_complete(self, action: syncSchemaConfig) -> bool:
+        """Check import action. Verify when sync tables from databricks is completed.
+
+        :param action:
+        :return:
+        """
+
+        public_schema = action["public_schema"]
+        table_name = action["table_name"]
+        schema = action["schema"]
+        if not self._dst_datastore.query(check_table_existence_query(public_schema, table_name)):
+            raise GOBException(f"Table '{public_schema}.{table_name}' does not exists.")
+        else:
+            try:
+                where_conditions = [{ "sync_schema": schema }]
+                select_query = create_select_where_query(table_name, where_conditions)
```

Comment:
Only the 3rd line in the try statement can possibly generate a StopIteration, move the rest out of the try/except